### PR TITLE
Feature 4

### PR DESCRIPTION
**Pull request by absolute-xero7**
Added a controller, and implemented CLEAN architecture. 
Added javadoc.
The design pattern I have used is the MVC design pattern. My controller is in the controllers package and my view is in the views package.
When this feature is run, a window is opened with the first exercise of the workout and empty text fields for the number of reps and the weight lifted. The next button starts a new view with the next exercise in the workout.

There are no tests in this pull request as the only 2 files are a controller and a view file.

### CODE REVIEW

**Review by bdls-jamal — DISMISSED** [review]
Any feedback from the previous pull on this feature(22) has been addressed or for positive feedback, kept the same SOLID principles.

File: src/main/java/controllers/WorkoutStartController.java [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+
+    }
+
+    /**
+     * Implements the behaviour of the next button.
+     */
+    class NextStartDisplay implements ActionListener {
+
+        @Override
+        public void actionPerformed(ActionEvent e) {
```
Comment (tejcapildeo) [inline src/main/java/controllers/WorkoutStartController.java:32]: Code smell in the form of a bloater. The method body is extremely huge. Maybe try extracting some of the if statements into another method, or find another work around. The code is hard to follow because of how big it is.

File: src/main/java/views/WorkoutStartView.java [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+
+        weight3.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                weights.add(weight3.getText());
+            }
+        });
+        reps3.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
```
Comment (tejcapildeo) [inline src/main/java/views/WorkoutStartView.java:149]: Methods in this file were kept short and readable.

**Review by tejcapildeo — COMMENTED** [review]
Comments were made within the document. The main issue was the lengths of the method bodies.

File: src/main/java/controllers/WorkoutStartController.java [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+                    }
+                    newStartView.setExerciseNumber(newStartView.getExerciseNumber()+1);
+                    newStartView.setExerciseName(startModel.getWorkouts()[2].getWorkout().get(newStartView.getExerciseNumber()).getType());
+
+                    new WorkoutStartController(startModel, newStartView);
+                }
+
+            }
+
+            if (startView.getWorkoutNumber() == 4){
```
Comment (ivyparc) [inline src/main/java/controllers/WorkoutStartController.java:127]: I'm seeing many duplicates in the codes since for each workout, the code bodies are very similar. Improvements could be made by having less copy of the body and have it run for each workout depending on something that indicates which workout it should run for. Overall nice implementation.

File: src/main/java/views/WorkoutStartView.java [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+    private JTextField reps1 = new JTextField();
+    private final JButton two = new JButton("2");
+    private JTextField weight2 = new JTextField();
+    private JTextField reps2 = new JTextField();
+    private final JButton three = new JButton("3");
+    private JTextField weight3 = new JTextField();
+    private JTextField reps3 = new JTextField();
+    private JButton ghost = new JButton();
+    private JButton next = new JButton("Next");
+    private JButton exercise = new JButton("");
```
Comment (ivyparc) [inline src/main/java/views/WorkoutStartView.java:41]: Good exercise of information of hiding by using private attributes.

File: src/main/java/views/WorkoutStartView.java [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+            }
+        });
+        reps2.addActionListener(new ActionListener() {
+            @Override
+            public void actionPerformed(ActionEvent e) {
+                repsNumber.add(reps2.getText());
+            }
+        });
+
+        weight3.addActionListener(new ActionListener() {
```
Comment (ivyparc) [inline src/main/java/views/WorkoutStartView.java:141]: Good practice of using Override. Also the methods are concise and readable.

**Review by ivyparc — COMMENTED** [review]
Comments are made within the document.
